### PR TITLE
TST: Avoid DeprecationWarnings

### PR DESCRIPTION
**Pull request by jbrockmendel**
Avoid a boatload of these:

```
pandas/tests/plotting/test_frame.py::TestDataFramePlots::()::test_plot
  /home/travis/build/pandas-dev/pandas/pandas/core/indexes/multi.py:1556: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
    if lab[key] == -1:

  /home/travis/build/pandas-dev/pandas/pandas/io/formats/style.py:1039: DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    return css_bar(min(x, zero), max(x, zero), colors[x > zero])

pandas/tests/series/test_analytics.py::TestCategoricalSeriesAnalytics::()::test_drop_duplicates_categorical_non_bool[False-datetime64[D]]
  source:2442: DeprecationWarning: invalid escape sequence \(
```

That last one was a nice little adventure tracking down to pd.util.testing.

Does not silence these, as I'm not confident that is The Right Thing To Do:
```
pandas/tests/tslibs/test_parsing.py::TestGuessDatetimeFormat::()::test_guess_datetime_format_nopadding[2011-1-1 0:0:0-%Y-%m-%d %H:%M:%S]
  /home/travis/miniconda3/envs/pandas/lib/python3.6/site-packages/dateutil/parser/__init__.py:46: DeprecationWarning: _timelex is a private class and may break without warning, it will be moved and or renamed in future versions.
    warnings.warn(msg, DeprecationWarning)
```

Does not silence these, as I'm not sure how:
```
pandas/tests/io/test_sql.py::TestSQLiteAlchemyConn::()::test_read_table
  /home/travis/miniconda3/envs/pandas/lib/python3.6/site-packages/_pytest/fixtures.py:795: RemovedInPytest4Warning: Fixture setup_method called directly. Fixtures are not meant to be called directly, are created automatically when test functions request them as parameters. See https://docs.pytest.org/en/latest/fixture.html for more information.
    res = next(it)
```

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -1551,6 +1551,10 @@ def __setstate__(self, state):
 
     def __getitem__(self, key):
         if is_scalar(key):
+            if lib.is_float(key) and key == int(key):
+                # Avoid DeprecationWarning for non-integer number indexer
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: we do this a number of times in the indexing code

i think we have a function

maybe_convert_to_int_indexer iirc

if not can u make one

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/multi.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: Index._convert_scalar_indexer looks like a candidate; need to check where it is called.

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: I take it back.  core.common is right next to `is_bool_indexer` is the place for this.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks! some more if you have a chance :>

**Pull request merged**